compiler/rustc_resolve/src/late.rs

```diff
@@ -4592,13 +4593,55 @@ impl<'ast> Visitor<'ast> for LifetimeCountVisitor<'_, '_, '_> {
 }
 
 impl<'a, 'tcx> Resolver<'a, 'tcx> {
-    pub(crate) fn late_resolve_crate(&mut self, krate: &Crate) {
+    pub(crate) fn late_resolve_crate(&mut self, krate: &mut Crate) {
         visit::walk_crate(&mut LifetimeCountVisitor { r: self }, krate);
-        let mut late_resolution_visitor = LateResolutionVisitor::new(self);
-        late_resolution_visitor.resolve_doc_links(&krate.attrs, MaybeExported::Ok(CRATE_NODE_ID));
-        visit::walk_crate(&mut late_resolution_visitor, krate);
-        for (id, span) in late_resolution_visitor.diagnostic_metadata.unused_labels.iter() {
-            self.lint_buffer.buffer_lint(lint::builtin::UNUSED_LABELS, *id, *span, "unused label");
+        {
+            let mut late_resolution_visitor = LateResolutionVisitor::new(self);
+            late_resolution_visitor
+                .resolve_doc_links(&krate.attrs, MaybeExported::Ok(CRATE_NODE_ID));
+            visit::walk_crate(&mut late_resolution_visitor, krate);
+            for (id, span) in late_resolution_visitor.diagnostic_metadata.unused_labels.iter() {
+                self.lint_buffer.buffer_lint(
+                    lint::builtin::UNUSED_LABELS,
+                    *id,
+                    *span,
+                    "unused label",
+                );
+            }
+        }
+
+        {
+            let mut late_desugar_visitor = LateDesugarVisitor::new(self);
+            late_desugar_visitor.visit_crate(krate);
+        }
+    }
+}
+
+struct LateDesugarVisitor<'a, 'b, 'tcx> {
+    r: &'b mut Resolver<'a, 'tcx>,
+}
+
+impl<'a, 'b, 'tcx> LateDesugarVisitor<'a, 'b, 'tcx> {
+    fn new(resolver: &'b mut Resolver<'a, 'tcx>) -> Self {
+        LateDesugarVisitor { r: resolver }
+    }
+}
+
+impl MutVisitor for LateDesugarVisitor<'_, '_, '_> {
+    fn visit_ty(&mut self, ty: &mut P<Ty>) {
+        if let TyKind::Path(None, _path) = &ty.kind
+            && let Some(partial_res) = self.r.partial_res_map.get(&ty.id)
+            && let Some(res @ Res::Def(DefKind::Trait | DefKind::TraitAlias, _)) =
+                partial_res.full_res()
+        {
+            debug!("[Desugar][Ty][BareTrait][{:?}] {:?}\n{:#?}", ty.span, res, ty);
+
+            let trait_obj = TyKind::TraitObject(vec![], TraitObjectSyntax::None);
```

Comment:
You are creating a trait object with a single generic bound, which is a trait with no bound generics (`for<'a>`) and no modifiers (`?Foo`, `const Bar`). The trait's path is `_path` from the type.